streams/src/main/java/org/apache/kafka/streams/kstream/Window.java

```diff
@@ -18,36 +18,56 @@
 package org.apache.kafka.streams.kstream;
 
 /**
- * A single window instance, defined by its start and end timestamp.
+ * A single window instance, defined by its start and end timestamp. TODO ?
  */
 public abstract class Window {
 
     private long start;
     private long end;
 
+    /**
+     * Create a new window for the given start time (inclusive) and end time (exclusive).
+     *
+     * @param start  the start timestamp of the window (inclusive)
+     * @param end    the end timestamp of the window (exclusive)
+     */
     public Window(long start, long end) {
         this.start = start;
         this.end = end;
     }
 
     /**
-     * Returns the start timestamp of this window, inclusive
+     * Return the start timestamp of this window, inclusive
      */
     public long start() {
         return start;
     }
 
     /**
-     * Returns the end timestamp of this window, exclusive
+     * Return the end timestamp of this window, exclusive
      */
     public long end() {
         return end;
     }
 
+    /**
+     * Check if the given window overlaps with this window.
+     *
+     * @param other  another window
+     * @return       {@code true} if {@code other} overlaps with this window&mdash;{@code false} otherwise
+     */
     public boolean overlap(Window other) {
         return this.start() < other.end() || other.start() < this.end();
     }
 
+    /**
+     * Check if the start and end timestamp of the given window are equal to the start and end timestamp
+     * of this window.
+     * 
+     * @param other  another window
+     * @return       {@code true} if start and end timestamp of {@code other} are equal to start end timestamp of this
```

Comment:
See typo fix above.

Nitpick: Also, why the `&mdash;`?  Typically you'd use a comma or semicolon (e.g. [Boolean.equals()](https://docs.oracle.com/javase/7/docs/api/java/lang/Boolean.html#equals%28java.lang.Object%29)):

> True if ...; false otherwise.